app/uk/gov/hmrc/rasapi/repository/RasFileRepository.scala

```diff
@@ -15,116 +15,115 @@
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 import uk.gov.hmrc.rasapi.config.AppContext
 import uk.gov.hmrc.rasapi.models.{Chunks, ResultsFile}
 
-import java.io.FileInputStream
-import java.nio.file.Path
+import java.nio.ByteBuffer
+import java.nio.file.{Files, Path}
 import javax.inject.Inject
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.{Duration, SECONDS}
+import scala.concurrent.{Await, ExecutionContext, Future}
 
 case class FileData(length: Long = 0, data: Enumerator[Array[Byte]] = Enumerator.empty)
 
-class RasFilesRepository @Inject()(val mongoComponent: ReactiveMongoComponent,
+class RasFilesRepository @Inject()(val mongoComponent: MongoComponent,
                                    val appContext: AppContext)
-                                   (implicit val ec: ExecutionContext) extends ReactiveRepository[Chunks, BSONObjectID](
+                                   (implicit val ec: ExecutionContext) extends PlayMongoRepository[Chunks](
+  mongoComponent = mongoComponent,
   collectionName = "rasFileStore",
-  mongo = mongoComponent.mongoConnector.db,
-  domainFormat = Chunks.format
+  domainFormat = Chunks.format,
+  indexes = Seq(),
+  replaceIndexes = false
 ) with GridFsTTLIndexing {
 
   override lazy val expireAfterSeconds: Long = appContext.resultsExpriyTime
-  private val contentType =  "text/csv"
   override val log: Logger = Logger(getClass)
-
-  val gridFSG: GridFS[BSONSerializationPack.type] =
-    GridFS[BSONSerializationPack.type](mongoComponent.mongoConnector.db(), "resultsFiles")
-
-  addAllTTLs(gridFSG)
-
-  def generateFileToSave(fileId: String, contentType: String, envelopeId: String, userId: String): FileToSave[BSONSerializationPack.type, BSONValue] =
-    DefaultFileToSave(
-      filename = Some(fileId),
-      contentType = Some(contentType),
-      uploadDate = Some(DateTime.now().getMillis),
-      metadata = BSONDocument("envelopeId" -> envelopeId, "fileId" -> fileId, "userId" -> userId)
-    )
+  private val contentType =  "text/csv"
+  private val bucketName: String = "resultsFiles"
+  val gridFSG: GridFSBucket = GridFSBucket(mongoComponent.database, bucketName)
 
   def saveFile(userId:String, envelopeId: String, filePath: Path, fileId: String): Future[ResultsFile] = {
     log.info("[RasFileRepository][saveFile] Starting to save file")
 
-    gridFSG.writeFromInputStream(generateFileToSave(fileId, contentType, envelopeId, userId), new FileInputStream(filePath.toFile)).map{ res =>
-      log.warn(s"Saved File id is ${res.id} for userId ($userId)")
-      res
-    }.recover{ case ex: Throwable =>
-        log.error(s"error saving file -> $fileId for userId ($userId). Exception: ${ex.getMessage}")
-        throw new RuntimeException(s"failed to save file due to error" + ex.getMessage)
+    val observableToUploadFrom: Observable[ByteBuffer] = Observable(
+      Seq(ByteBuffer.wrap(Files.readAllBytes(filePath).array)
```

Comment:
do you need to call the underyling `.array` here? it seems to compile without it